### PR TITLE
Add support for Python 3.11

### PR DESCRIPTION
**Pull request by harahu**
Now that Python 3.11 is released, it is time to start officially supporting it.

### CODE REVIEW

**Comment by harahu** [comment]
@francium Would you mind having a look at merging this? I don't think this one should be very controversial.

File: pyproject.toml [hunk 1]
```diff
@@ -24,7 +24,6 @@ no_implicit_reexport = true
 show_column_numbers = true
 show_error_codes = true
 show_error_context = true
-show_none_errors = true
```
Comment (wbolster) [inline pyproject.toml:27]: 🤔  this is the default behaviour these days, no?

**Review by wbolster — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (wbolster) [reply pyproject.toml:27]: ah it's old stuff: https://github.com/python/mypy/pull/13507

**Comment by wbolster** [comment]
thanks!

**Pull request merged**